### PR TITLE
Fixes #11914: Add Rudder reporting context before calling technique bundle

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/11914

### CODE REVIEW

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/write/BuildBundleSequence.scala [hunk 1]
```diff
@@ -146,8 +152,10 @@ object BuildBundleSequence {
     , isSystem               : Boolean
     , providesExpectedReports: Boolean
     , policyMode             : PolicyMode
+    , bundleContext          : Option[BundleContext]
```
Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/write/BuildBundleSequence.scala]: Why an option ? We always have that information. Is there some case where we don't want to use it?

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/write/BuildBundleSequence.scala [hunk 2]
```diff
@@ -118,6 +120,10 @@ object BuildBundleSequence {
   // The rudder id is ruleid@@directiveid@@0
   final case class ReportId(value: String)
```
Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/write/BuildBundleSequence.scala]: Isn't reportId the same thing that BundleContext ?

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/write/BuildBundleSequence.scala (hunk 1 above)
Comment (VinceMacBuche) [reply rudder-core/src/main/scala/com/normation/rudder/services/policies/write/BuildBundleSequence.scala]: I don't think we want to define context for all Bundles ( We only want it for the main Bundle, not the pre and the post ones) 

But maybe we could add the bundle in the right place without having to set this ... I'll look if i find a better way

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/write/BuildBundleSequence.scala (hunk 2 above)
Comment (VinceMacBuche) [reply rudder-core/src/main/scala/com/normation/rudder/services/policies/write/BuildBundleSequence.scala]: The identifier produced at the end is the report id, But i wanted to build it agent side by sending ruleId and directiveid separately so we can use then more easily in the technique ('maybe add debug / verbose info ...

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
PR rebased

**Comment by VinceMacBuche** [comment]
Commit modified

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/write/BuildBundleSequence.scala [hunk 3]
```diff
@@ -354,8 +356,7 @@ object CfengineBundleVariables extends AgentFormatBundleVariables {
         val newMain = tb.main match {
           case Nil => Nil
           // in that case, we are using the same ReportId for report and bundle
-          case firstBundle :: tail => (Bundle(firstBundle.id, BundleName(s"""current_technique_report_info("${firstBundle.name.value}")"""), Seq())
-                                       :: firstBundle :: tail)
+          case firstBundle :: tail =>  firstBundle :: tail
```
Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/write/BuildBundleSequence.scala]: i fail to see what's the benefit of this pattern matching

Comment (VinceMacBuche) [reply rudder-core/src/main/scala/com/normation/rudder/services/policies/write/BuildBundleSequence.scala]: You are totally right, just forgot to clean after removing what need to be removed

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
OK, merging this PR

**Pull request merged**